cranelift/isle/isle/src/ast.rs

```diff
@@ -419,3 +420,26 @@ pub enum ArgPolarity {
     /// *from* the extractor op.
     Output,
 }
+
+/// An implicit converter, which can appear in extractor or
+/// constructor position. This sets up an ambient automatic type
+/// conversion that is used when needed.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Converter {
+    /// The Rust function name.
+    pub func: Ident,
+    /// The type to convert *from*.
+    pub from_ty: Ident,
+    /// The type to convert *to*.
+    pub to_ty: Ident,
+    /// Infallibility: if an external converter returns `ToType`
+    /// rather than `Option<ToType>` hence can never fail, it is
+    /// declared as such and allows for slightly better code to be
+    /// generated.
+    pub infallible: bool,
```

Comment:
Can this be inferred from whether the term used for converting is infallible?

Reply:
Yes, if we go to a term-based design instead, for sure.